marlowe-playground-client/src/Simulation.purs

```diff
@@ -132,16 +135,26 @@ handleAction _ (SetEditorText contents) = do
   editorSetValue contents
   updateContractInState contents
 
+handleAction _ (SetInitialSlot initialSlot) = do
+  assign (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _initialSlot) initialSlot
+
 handleAction _ StartSimulation = do
-  assign (_currentMarloweState <<< _executionState) (Just $ emptyExecutionStateWithSlot zero)
-  moveToSlot zero
+  maybeInitialSlot <- peruse (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _initialSlot)
+  case maybeInitialSlot of
```

Comment:
one pattern I've been using a bit these days is when you have an action you want to take with a `Just value` and do nothing otherwise. In this case instead of pattern matching you can do
```purescript
for_ maybeInitialSlot \initialSlot -> do
...
```
It's just a matter of style but putting it in this comment in case you think it would be nice here

Reply:
ah, makes sense 👍